### PR TITLE
[item] fix lazy loading for reindex

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
Removed cross join since it was fetching all items at once

### CODE REVIEW

File: app/Elasticsearch/Repositories/ItemRepository.php [hunk 1]
```diff
@@ -445,27 +445,28 @@ public function reindexAllLocales(Carbon $viewedSince = null): int
 
         $query
             ->lazy()
-            ->tap(function () use (&$processedCount) {
+            ->tapEach(function () use (&$processedCount) {
```
Comment (eronisko) [inline app/Elasticsearch/Repositories/ItemRepository.php:448]: 👍

File: app/Elasticsearch/Repositories/ItemRepository.php [hunk 2]
```diff
@@ -445,27 +445,28 @@ public function reindexAllLocales(Carbon $viewedSince = null): int
...
+                    ];
 
                     // Data
-                    $item->getIndexedData($locale),
-                ];
+                    $operations[] = $item->getIndexedData($locale);
+                }
+
+                return $operations;
             })
```
Comment (eronisko) [inline app/Elasticsearch/Repositories/ItemRepository.php:468]: Doesn't this do the same thing as before?

File: app/Elasticsearch/Repositories/ItemRepository.php [hunk 3]
```diff
@@ -445,27 +445,28 @@ public function reindexAllLocales(Carbon $viewedSince = null): int
...
-            ->chunk(1000)
+            ->chunk(2 * count($this->locales) * 200)
```
Comment (eronisko) [inline app/Elasticsearch/Repositories/ItemRepository.php:469]: Oh, did this split the actions and data?

**Review by eronisko — COMMENTED** [review]
What was the problem/reason for this change?

File: app/Elasticsearch/Repositories/ItemRepository.php (hunk 3 above)
Comment (rastislav-chynoransky) [reply app/Elasticsearch/Repositories/ItemRepository.php:469]: No, it just has to be divisible by 2, but this seemed to me more readable what the number means

File: app/Elasticsearch/Repositories/ItemRepository.php (hunk 2 above)
Comment (rastislav-chynoransky) [reply app/Elasticsearch/Repositories/ItemRepository.php:468]: It removes cross join since it was fetching all items at once to build it

Comment (eronisko) [reply app/Elasticsearch/Repositories/ItemRepository.php:468]: Oh, I see 😬 thank you

File: app/Elasticsearch/Repositories/ItemRepository.php (hunk 3 above)
Comment (eronisko) [reply app/Elasticsearch/Repositories/ItemRepository.php:469]: Could you please add a comment explaining the formula? (It was not obvious to me)

**Review by eronisko — APPROVED** [review]
Thank you for this. I believe this'll also fix https://jira.sng.sk/browse/WEBUMENIA-1887

**Comment by rastislav-chynoransky** [comment]
> Thank you for this. I believe this'll also fix https://jira.sng.sk/browse/WEBUMENIA-1887

Oh, I didn't know about that one. Yes, I think you can close it.

**Pull request merged**